Add a convention about z_index (#9701)
minetest/minetest: doc/lua_api.txt

The `z_index` field specifies the order of HUD elements from back to front.
Lower z-index elements are displayed behind higher z-index elements. Elements
with same z-index are displayed in an arbitrary order. Default 0.
Supports negative values. By convention, the following values are recommended:

*  -400: Graphical effects, such as vignette
*  -300: Name tags, waypoints
*  -200: Wieldhand
*  -100: Things that block the player's view, e.g. masks
*     0: Default. For standard in-game HUD elements like crosshair, hotbar,
         minimap, builtin statbars, etc.
*   100: Temporary text messages or notification icons

> *  1000: Full-screen effects such as full-black screen or credits.

Reviewer: How about a smaller number?

The `z_index` field specifies the order of HUD elements from back to front.
Lower z-index elements are displayed behind higher z-index elements. Elements
with same z-index are displayed in an arbitrary order. Default 0.
Supports negative values. By convention, the following values are recommended:

*  -400: Graphical effects, such as vignette
*  -300: Name tags, waypoints
*  -200: Wieldhand
*  -100: Things you wear on your face or stuff that is blocking your view,

>          such as a mask

Reviewer 2: In general, I think direct speech should be avoided (-> `you`).
For example: `Elements that block the player's view, e.g. masks.`

*     0: Default. For builtin HUD elements and anything that is
         considered to be the “true” HUD:

>          Minimap, hotbar, builtin statbars, crosshair

Reviewer 2: How about:
`Standard in-game HUD elements that are usually visible: Minimap, hotbar...`

*   100: Temporary text messages or notification icons
*  1000: Full-screen effects such as full-black screen or credits.

>          Basically anything that must cover the entire screen and
>          replace the entire HUD

Reviewer 2: ```suggestion
         This includes effects that cover the entire screen.
```
